Main branch broken due to handling user-defined imports

When running the tests of the main branch they fail at:
```
[ERROR] Failures:
[ERROR] org.mapstruct.ap.test.bugs._2897.Issue2897Test.shouldImportNestedClassInMapperImports
[ERROR]   Run 1: Issue2897Test.shouldImportNestedClassInMapperImports [Expected checkstyle compliant output, but got errors:
[ERROR] C:\Programmeren\github\mapstruct\processor\target\compilation-tests\org.mapstruct.ap.test.bugs._2897.Issue2897Test\shouldImportNestedClassInMapperImports_jdk\generated-sources\org\mapstruct\ap\test\bugs\_2897\Issue2897MapperImpl.java:4:8: Nicht verwendeter Import - org.mapstruct.ap.test.bugs._2897.util.Util. [UnusedImports]
]
Expecting:
 <true>
to be equal to:
 <false>
but was not.
[ERROR]   Run 2: Issue2897Test.shouldImportNestedClassInMapperImports [Expected checkstyle compliant output, but got errors:
[ERROR] C:\Programmeren\github\mapstruct\processor\target\compilation-tests\org.mapstruct.ap.test.bugs._2897.Issue2897Test\shouldImportNestedClassInMapperImports_eclipse\generated-sources\org\mapstruct\ap\test\bugs\_2897\Issue2897MapperImpl.java:4:8: Nicht verwendeter Import - org.mapstruct.ap.test.bugs._2897.util.Util. [UnusedImports]
]
Expecting:
 <true>
to be equal to:
 <false>
but was not.
```

I'm creating a fix for this.

--- a/processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java
+++ b/processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java
@@ -181,9 +181,16 @@ public class Type extends ModelElement implements Comparable<Type> {
 
         this.loggingVerbose = loggingVerbose;
 
-        // The top level type for an array type is the top level type of the component type
-        TypeElement typeElementForTopLevel =
-            this.componentType == null ? this.typeElement : this.componentType.getTypeElement();
+        TypeElement typeElementForTopLevel;
+        if ( Boolean.TRUE.equals( isToBeImported ) ) {
+            // If the is to be imported is explicitly set to true then we shouldn't look for the top level type
+            typeElementForTopLevel = null;
+        }
+        else {
+            // The top level type for an array type is the top level type of the component type
+            typeElementForTopLevel =
+                this.componentType == null ? this.typeElement : this.componentType.getTypeElement();
+        }
         this.topLevelType = topLevelType( typeElementForTopLevel, this.typeFactory );
         this.nameWithTopLevelTypeName = nameWithTopLevelTypeName( typeElementForTopLevel, this.name );
     }

